src/main/java/com/facethat/resources/HomeResources.java

```diff
@@ -1,55 +1,94 @@
 
-  private final String apiKey;
+    private final WeatherDao weatherDao;
 
-  public HomeResources(String template, String defaultName, String apiKey) {
-    this.template = template;
-    this.defaultName = defaultName;
-    this.apiKey = apiKey;
-  }
+    public HomeResources(String template, String defaultName, String apiKey, WeatherDao weatherDao) {
+        this.template = template;
+        this.defaultName = defaultName;
+        this.apiKey = apiKey;
+        this.weatherDao = weatherDao;
+    }
 
 
-  @GET
-  public Response  getHomeDetails(){
+    @GET
+    public Response getHomeDetails() {
 
-    return Response
-        .ok(defaultName+"\n"+template+" This is a  simple weather App")
-        .build();
-  }
+        return Response.ok(defaultName + "\n" + template + " This is a  simple weather App")
+                       .build();
+    }
 
-  @GET
-  @Path("weather")
-  public  Response getAClient(){
+    @GET
+    @Path("/weather/{q}")
+    public Response getWeather(@QueryParam("city") String city) throws JsonProcessingException {
 
-    Client client  = ClientBuilder.newClient();
 
-    Object result = client.
-        target("https://api.weatherapi.com/v1/current.json?key="+apiKey+"&q=lagos")
-        .request().get(Object.class);
+        Current dbCity = weatherDao.getCity(city);
 
-    return Response
-        .ok(result)
-        .build();
+        ///Get the time a weather condition of a place is saved into the database
+        String timeSavedToDatabase = weatherDao.getTimeSavedToDatabase(city);
+        //Convert timeSavedToDatabase into local_day_time
+        LocalDate localDateTime = LocalDate.parse(timeSavedToDatabase);
 
 
-  }
+        ////The time limit which a weather data condition can stay in the database
+        LocalDate timeLimit = LocalDate.now().plusDays(1);
+
+
+        /// if the time is greater than a day, the city data should be deleted
+        if (LocalDate.now().plusDays(1).isAfter(timeLimit)) {
+            weatherDao.deleteCity(city);
+        }
+
+        // Return weather data, if the data exist in the database
+        if (dbCity!=null && !localDateTime.equals(timeLimit)) {
+
+            return Response.ok(dbCity).build();
+
+        } else {
+            Client client = ClientBuilder.newClient();
+            ///Fetch Weather data form remote api
+            Weather result = client.target("https://api.weatherapi.com/v1/current.json?key=" + apiKey + "&q=" + city + "")
+                                   .request().get(Weather.class);
+
+            Current current = result.getCurrent();
+
+            ///Converts Weather  object to Json (as String)
+            ObjectMapper mapper = new ObjectMapper();
+            String convertedObject = mapper.writeValueAsString(result);
+
+
+            ///Convert Json(as String) String to weather Json
+            Weather weather = mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false).readValue(convertedObject, Weather.class);
```

Comment:
Why are we de-serializing weather twice
- Weather result = ...
- Weather weather = mapper.configre(...)
- Also can't we just return the domain object (`Weather.class`) instead of `Response`, dropwizard can generate a json response from your domain object.